packages/web-components/fast-ssr/src/dom-shim.ts

```diff
@@ -119,14 +119,48 @@ class CustomEvent<T = any> extends Event {
     }
 }
 
+class CSSStyleRule {
+    public readonly style = new CSSStyleDeclaration();
+    constructor(public selectorText: string) {}
+
+    public get cssText(): string {
+        return `${this.selectorText} { ${this.style.cssText} }`;
+    }
+}
+
+class CSSStyleDeclaration {
+    private rules = new Map<string, string>();
+    public setProperty(name: string, value: string = "") {
+        this.rules.set(name, value);
+    }
+    public removeProperty(name: string) {
+        this.rules.delete(name);
+    }
+
+    public get cssText() {
+        let text = "";
+
+        for (const [key, value] of this.rules) {
+            text += `${key}: ${value};`;
+        }
+
+        return text;
+    }
+}
+
 /**
  * @beta
  */
 export class CSSStyleSheet {
     replace() {}
     public readonly cssRules: CSSRule[] = [];
-    insertRule(rule: CSSRule, index: number = 0) {
-        this.cssRules.splice(index, 0, rule);
+    insertRule(rule: string, index: number = 0) {
+        const selectorText = rule.split("{")[0];
```

Comment:
Is this safe for media queries?

Reply:
No. This shim isn't supposed to be comprehensive, and there are many APIs that are incomplete or don't exist on these objects. Is there a core case we need to handle that includes writing media queries w/ insertRule?

Reply:
> Is there a core case we need to handle that includes writing media queries w/ insertRule?

Not that I'm familiar with - I just like to double-check when I see string manipulation used to assist code evaluation.

Reply:
Gotcha, sounds good. The shim is extensible for anyone that needs a more robust implementation of these APIs (or they can choose to use a different shim all together) - the goal of *this* shim is to cover the API usage of fast-element and fast-foundation.